couchbase/metrics/logging_meter.cxx

```diff
@@ -171,14 +171,16 @@ logging_meter::get_value_recorder(const std::string& name, const std::map<std::s
     if (service == tags.end()) {
         return &noop_recorder;
     }
-    auto& service_recorders = recorders_[service->second];
 
     static std::string operation_tag = "db.operation";
     auto operation = tags.find(operation_tag);
     if (operation == tags.end()) {
         return &noop_recorder;
     }
 
+    std::scoped_lock lock(recorders_mutex_);
+    auto& service_recorders = recorders_[service->second];
+
     auto recorder = service_recorders.find(operation->second);
```

Comment:
Looks reasonable, although I noticed below this section there is a bit of weirdness in the implementation which will likely have a performance impact.  Will file an issue for it.

Reply:
https://issues.couchbase.com/browse/CXXCBC-75

Reply:
https://issues.couchbase.com/browse/CXXCBC-76